File: android/activitysources/src/main/java/com/fjuul/sdk/activitysources/entities/internal/healthconnect/HCHeartRateDataPoint.java
package com.fjuul.sdk.activitysources.entities.internal.healthconnect;

import java.util.Date;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public abstract class HCHeartRateDataPoint extends HCDataPoint {
    private final float heartRate;

    protected HCHeartRateDataPoint(@NonNull Date start, @Nullable Date end, float heartRate, @Nullable String dataSource) {
        super(start, end);
        this.heartRate = heartRate;
        if (dataSource != null) {
            addDataSource(dataSource);
        }
    }

    public float getHeartRate() {
        return heartRate;
    }
} 

File: android/activitysources/src/main/java/com/fjuul/sdk/activitysources/entities/internal/healthconnect/HCHeartRateDataPoint.kt
package com.fjuul.sdk.activitysources.entities.internal.healthconnect

import java.time.Instant

/**
 * Represents a heart rate record from Health Connect.
 */
data class HCHeartRateDataPoint(
    override val start: Instant,
    override val value: Double,
    override val dataSources: Set<String>
) : HCIntradayDataPoint()

/**
 * Represents a resting heart rate record from Health Connect.
 */
data class HCRestingHeartRateDataPoint(
    override val start: Instant,
    override val value: Double,
    override val dataSources: Set<String>
) : HCDailyDataPoint()
